### PR TITLE
Add types for @tgwf/co2

### PR DESCRIPTION
**Pull request by fershad**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/tgwf__co2/package.json
1. types/tgwf__co2/tsconfig.json
1. types/tgwf__co2/tgwf__co2-tests.test.ts
1. types/tgwf__co2/index.d.ts
1. types/tgwf__co2/co2.d.ts

<details>
<summary>as well as these 4 other files...</summary>
<p>types/tgwf__co2/tgwf__co2-tests.ts, types/tgwf__co2/1byte.d.ts, types/tgwf__co2/hosting.d.ts, types/tgwf__co2/sustainable-web-design.d.ts</p>
</details>


Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against bc9c6ad4a62ee41ef874e0fd6850b35dcfdc67ba
</p>

**Comment by peterblazejewicz** [comment]
It seems there is ongoing effort in upstream project to introduce types declaration to the module:
1) there is blanked declaration:
https://github.com/thegreenwebfoundation/co2.js/blob/main/index.d.ts
2) dedicated issue:
https://github.com/thegreenwebfoundation/co2.js/issues/77
what's relation of this PR to the community efforts? thx!

**Comment by fershad** [comment]
Thanks @peterblazejewicz. The maintainers (@mrchrisadams and I) have no plans to convert the library to Typescript. https://github.com/thegreenwebfoundation/co2.js/issues/77 was raised by Chris to gain an understanding as to what my be involved in the process. In subsequent conversations, some of which have been internal to our team and aren't captured in the issue at present, we've decided not to convert the library to Typescript. Chris and I as maintainers lack the TS know-how to ensure this would be done properly.

The empty `index.d.ts` file was created in https://github.com/thegreenwebfoundation/co2.js/pull/131 to stop TS users from getting an error when using the library in their projects. 

When this PR (this one we're in now) merges, we would remove that `index.d.ts` file and advise all users to get import type definitions from this source instead.

We feel that having type definitions in this open source repo will allow those who need them to pull them into their projects without imposing them on everyone.

**Comment by mrchrisadams** [comment]
hi folks I can confirm that we do not have plans to convert the library to Typescript and the Definitely Typed approach is the one we prefer for the co2.js library. I have one query though:

> And if you do find someone, maybe even recruit them to be a second owner to make future changes easier...

I'm not sure how to become a second owner, but I'm happy to do so, once I know how.

**Comment by peterblazejewicz** [comment]
> I'm not sure how to become a second owner, but I'm happy to do so, once I know how.


just amend the owners with GitHub handle and give an ACK to the change:
```json
"owners": [
    {
        "name": "Fershad Irani",
        "githubUsername": "fershad"
    }
]
```

**Comment by fershad** [comment]
I've added @mrchrisadams as an owner.

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/tgwf__co2/package.json

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c1b6b3ff9cb8806f4b360a34b3c39572620a5b57
</p>

File: types/tgwf__co2/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    export { marginalIntensity };
+}
+
+import co2 from "./co2.d.ts";
+import averageIntensity from "./data/average-intensities.min.d.ts";
+import marginalIntensity from "./data/marginal-intensities-2021.min.d.ts";
+import hosting from "./hosting.d.ts";
+export { averageIntensity, co2, hosting, marginalIntensity };
+
+export default co2js;
```
Comment (sandersn) [inline types/tgwf__co2/index.d.ts:14]: unfortunately, DT still technically only provides types for commonjs -- so you need to use `export = co2js` here, which is the Typescript syntax for commonjs exports. `export default co2js` is only allowed for commonjs when the commonjs version exports a default property like so: `require('@tgwf/co2').default`. But @tgwf/co2 doesn't do this.
```suggestion
export = co2js;
```

You can hack in types for ES exports, but it's still janky. See `types/three` for an example.

File: types/tgwf__co2/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "paths": {
+            "@tgwf/co2": [
+                "./index.d.ts"
+            ]
+        },
```
Comment (sandersn) [inline types/tgwf__co2/tsconfig.json:11]: this should not be needed. Can you try deleting it?

**Review by sandersn — CHANGES_REQUESTED** [review]

**Comment by fershad** [comment]
Thanks for the pointers @sandersn. I've made those changes [14f9b1c](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67910/commits/14f9b1cc01c5ff627726fc41ac5965a4556c1ced) and [5c8857f](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67910/commits/5c8857ffdc1108ba5d2105feb255f618bdf05b54)

**Comment by fershad** [comment]
Apologies, a few more updates after tests failed previously.

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/tgwf__co2/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f25b9076c71d7000eb671b7596074a18da00ad6d
</p>

**Review by sandersn — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by fershad** [comment]
Ready to merge

**Comment by fershad** [comment]
Thanks all for your help in getting this merged. Really appreciate the review & guidance.

**Pull request merged**